kpatch-build/create-diff-object.c

```diff
@@ -2642,6 +2707,22 @@ static void kpatch_regenerate_orc_sections(struct kpatch_elf *kelf)
 	ip_sec->data->d_size = dest_idx * ORC_IP_PTR_SIZE;
 }
 
+static bool skip_check_relocation(Elf_Data *sdata, struct rela *rela)
+{
+	/*
+	 * When CONFIG_LOCKDEP is disabled, struct lock_class_key is of size
+	 * zero. Pointer __key of type lock_class_key is passed to
+	 * __init_waitqueue_head(), eventhough the pointer is not used.
+	 * Ignore out-of-range relocation in this case.
+	 */
+	if (!sdata->d_size &&
+	    (!strncmp(rela->sym->name, "__key.", 6) ||
+	    !strncmp(rela->sym->name, ".bss.__key.", 11)))
+		return true;
+	else
+		return false;
```

Comment:
It would work. But it would skip valid ones like:
rela->sym->name: __func__.0 sec->name: .rela.text.audit_log_start.part.0 rela->sym->sec->sh.sh_size: 16 sdata->size: 16  rela->sym->sym.st_value: 0,

Should we check for sym->type == STT_OBJECT && !sdata->d_size instead ?

Reply:
Sorry, typo.  I meant to check the symbol size, something like `sym->type == STT_OBJECT && !sym->sym.st_size`

Reply:
Added